docs/proposals/karmadactl/Karmadactl Supports Configuration File Method for Production-Grade Installation and Deployment.md

```diff
@@ -0,0 +1,367 @@
+
+For the design of the configuration file, I think it can be designed similarly to the config structure in `kubeadm init`. I will provide a YAML file example below. To provide users with a configuration file, I think we can prepare a default configuration file example and make it a sub-command in `karmadactl config`. When the feature design is complete and the official documentation is updated, an example can also be demonstrated in the official documentation.
+
+Example configuration file:
+
+```yaml
+apiVersion: config.karmada.io/v1beta1
+kind: InitConfiguration
+general:
+  namespace: "karmada-system"
+  kubeConfigPath: "/etc/karmada/kubeconfig"
+  kubeImageTag: "v1.21.0"
+  context: "karmada-context"
+  storageClassesName: "fast"
+  privateImageRegistry: "local.registry.com"
+  waitComponentReadyTimeout: 120
+  port: 32443
+
+certificate:
+  externalDNS:
+    - "www.karmada.io"
+  externalIP:
+    - "10.235.1.2"
+  validityPeriod: "8760h"
+  caCertFile: "/etc/karmada/pki/ca.crt"
+  caCertKeyFile: "/etc/karmada/pki/ca.key"
+  extraArgs:
+    - name: "arg1"
+      value: "value1"
+
+etcd:
+  local:
+    image: "local.registry.com/library/etcd:3.5.13-0"
+    initImage: "docker.io/alpine:3.19.1"
+    dataDir: "/var/lib/karmada-etcd"
+    pvcSize: "5Gi"
+    nodeSelectorLabels:
+      kubernetes.io/os: linux
+      hello: world
+    storageMode: "PVC"
+    replicas: 3
+    extraArgs:
+      - name: "etcd-arg1"
+        value: "etcd-value1"
+  external:
+    caPath: "/etc/ssl/certs/ca-certificates.crt"
+    certPath: "/path/to/your/certificate.pem"
+    keyPath: "/path/to/your/privatekey.pem"
+    servers: "https://example.com:8443"
+    prefix: "ext-"
+    extraArgs:
+      - name: "ext-etcd-arg1"
+        value: "ext-etcd-value1"
+
+karmadaControlPlane:
+  apiServer:
+    image: "karmada-apiserver:latest"
+    advertiseAddress: "192.168.1.2"
+    replicas: 3
+    nodePort: 32443
+    extraArgs:
+      - name: "api-arg1"
+        value: "api-value1"
+  controllerManager:
+    image: "karmada-controller-manager:latest"
+    replicas: 3
+    extraArgs:
+      - name: "cm-arg1"
+        value: "cm-value1"
+  scheduler:
+    image: "karmada-scheduler:latest"
+    replicas: 3
+    extraArgs:
+      - name: "sched-arg1"
+        value: "sched-value1"
+  webhook:
+    image: "karmada-webhook:latest"
+    replicas: 3
+    extraArgs:
+      - name: "webhook-arg1"
+        value: "webhook-value1"
+  aggregatedAPIServer:
+    image: "karmada-aggregated-apiserver:latest"
+    replicas: 3
+    extraArgs:
+      - name: "aggregated-arg1"
+        value: "aggregated-value1"
+  kubeControllerManager:
+    image: "kube-controller-manager:latest"
+    replicas: 3
+    extraArgs:
+      - name: "kcm-arg1"
+        value: "kcm-value1"
+  dataPath: "/etc/karmada"
+  pkiPath: "/etc/karmada/pki"
+  crds: "/etc/karmada/crds"
+  hostClusterDomain: "cluster.local"
+
+image:
```

Comment:
It seems that this section can be placed under `general`

```
general:
  image:
```

Reply:
I've put the image in general, and I've completed a few instances, so how does it look now